### PR TITLE
Adds a Hackernews example site

### PR DESCRIPTION
**Pull request by tony-sull**
## Changes

- Adds a new `hackernews` example starter

[Live preview](https://astro-hackernews.netlify.app)

## Testing

Tested manually, all CI tests should pass

## Docs

Includes a README with details on the project structure and use of SSR

/cc @withastro/maintainers-docs

### CODE REVIEW

File: examples/hackernews/README.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+# Welcome to [Astro](https://astro.build) Hackernews
```
Comment (sarah11918) [inline examples/hackernews/README.md:1]:
```suggestion
# Astro Starter Kit: Hackernews

```
npm create astro@latest -- --template hackernews
```
```
 Just pointing out that there's another open PR #5201 that is standardizing some of the example READMEs.

Comparing to that, I think the only thing to keep this one in line with that model is updating the title to the  `Astro Starter Kit: Basics` with create astro command format.

File: examples/hackernews/README.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+## 🚀 Project Structure
+
+Inside of your Astro project, you'll see the following folders and files:
+
+```
+/
+├── public/
+│   └── favicon.svg
+├── src/
+│   ├── components/
```
Comment (sarah11918) [inline examples/hackernews/README.md:16]: Just checking that it was intentional to leave out the component files here?

File: examples/hackernews/README.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+All commands are run from the root of the project, from a terminal:
+
+| Command                | Action                                             |
+| :--------------------- | :------------------------------------------------- |
+| `npm install`          | Installs dependencies                              |
+| `npm run dev`          | Starts local dev server at `localhost:3000`        |
+| `npm run build`        | Build your production site to `./dist/`            |
+| `npm run preview`      | Preview your build locally, before deploying       |
+| `npm run astro ...`    | Run CLI commands like `astro add`, `astro preview` |
```
Comment (sarah11918) [inline examples/hackernews/README.md:48]: There's a short discussion on the other PR about `astro check` being used as an example, but it's not always included in every project. `astro docs` is another option. I'm just wondering whether we want `astro preview` here in addition to `npm run preview`, or whether a different one might add some variety!

**Review by sarah11918 — COMMENTED** [review]
The system works!  🥳 

I enjoyed checking out the site, Tony! Left a few comments on the README for your consideration.

File: examples/hackernews/src/components/For.astro [hunk 4]
```diff
@@ -0,0 +1,21 @@
+---
```
Comment (tony-sull) [inline examples/hackernews/src/components/For.astro:1]: :eyes: flow components :eyes:

Not totally sold on using these in one of our example starters, happy to swap these back out for plain JSX expressions if anyone has concerns

File: examples/hackernews/README.md (hunk 3 above)
Comment (tony-sull) [reply examples/hackernews/README.md:52]: Yeah we should probably clean up the examples to all use the same scripts, I just copied these over from `basics` but do like the `"check": "astro check && tsc",` script included in `docs`

Comment (jasikpark) [reply examples/hackernews/README.md:52]: I've been enjoying using
```
"check:astro": "astro check",
"check:ts": "tsc -p .",
"check": "run-p check:*",
```

Comment (tony-sull) [reply examples/hackernews/README.md:52]: Yeah I actually use `npm-run-all` in most projects too, I think we skipped it in starters just to keep the dependencies list down

**Pull request merged**